mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala

```diff
@@ -139,16 +145,32 @@ class KMeansSuite extends SparkFunSuite with MLlibTestSparkContext with DefaultR
     val kmeans = new KMeans()
     testEstimatorAndModelReadWrite(kmeans, dataset, KMeansSuite.allParamSettings, checkModelData)
   }
+
+  test("Initialize using given cluster centers") {
```

Comment:
IMO, the best test here is to train a model on some sample dataset. Then train a model using that initial model on the same dataset, and verify that it converges in a single iteration. I don't think we have a way to access the number of iterations, so that isn't possible at the moment (unless I'm missing it?). Can we add a TODO that when (if?) we add total iterations to the KMeans model summary we can change this test?